python/taichi/lang/ir_builder.py

```diff
@@ -146,7 +161,31 @@ def build_Call(ctx, node):
             else:
                 args.append(arg.ptr)
         keywords = dict(ChainMap(*[keyword.ptr for keyword in node.keywords]))
-        node.ptr = node.func.ptr(*args, **keywords)
+
+        if isinstance(node.func, ast.Attribute):
+            attr_name = node.func.attr
+            if attr_name == 'format':
+                args.insert(0, node.func.value.ptr)
+                node.ptr = ti.ti_format(*args, **keywords)
+        if isinstance(node.func, ast.Name):
```

Comment:
if->elif, and add an else to this one too

Reply:
is this one resolved?

Reply:
resolved now